PROTOCOL.md

```diff
@@ -0,0 +1,75 @@
+
+Invitations are published into invite topics. Each blockchain account is automatically assigned an invite topic named `invite-<account address>`. It contains all the invitations both sent and received by the account owner. It allows the client to reconstruct the list of all past conversations and to find their corresponding message topics.
+
+Invitations are encrypted using a key derived from the pre-keys of the invitation sender and recipient. Consequently the sender and the recipient of an invitation are present in the header, outside of the encrypted payload, and visible to outside observers (i.e. nodes).
+
+### Messages
+
+[Messages](https://github.com/xmtp/proto/blob/main/proto/message_contents/xmtp_envelope.proto) are published into conversation topics indicated by the corresponding invitation. Conversation topic names follow the format `m-<base64 encoded random 32bytes>`.
+
+Message content is encoded using a content type framework represented by the [EncodedContent type](https://github.com/xmtp/proto/blob/main/proto/message_contents/xmtp_envelope.proto). Usage of the content type framework is governed by [XIP-5](https://github.com/xmtp/XIPs/blob/main/XIPs/xip-5-message-content-types.md).
+
+The bytes of the `EncodedContent` are wrapped in [SignedContent](https://github.com/xmtp/proto/blob/main/proto/message_contents/xmtp_envelope.proto), signed using the pre-key of the sender. The signature covers the bytes of the `EncodedContent` and the bytes of the [MessageHeaderV2](https://github.com/xmtp/proto/blob/main/proto/message_contents/xmtp_envelope.proto) to frustrate replay attacks. `SignedContent` is then encrypted using the key material from the invitation and wrapped in [Ciphertext](https://github.com/xmtp/proto/blob/main/proto/message_contents/ciphertext.proto). Finally [MessageV2](https://github.com/xmtp/proto/blob/main/proto/message_contents/xmtp_envelope.proto) combines the `Ciphertext` and the bytes of the message header.
+
+## V1
+
+Previous version of the protocol (further V1) had the differences described below.
+
+The networks support both versions of the protocol in order to allow retrieving historic conversations correctly. V1 conversations can still receive new messages, if the parties initiated them before V2 protocol rollout.
+
+### Keys
+
+The protocol used a different format for keys and key bundles, see [PublicKeyBundle](https://github.com/xmtp/proto/blob/main/proto/message_contents/public_key.proto), although semantically there was very little difference. The old format had optional public key signatures, although this was never actually the case, the signatures were effectively required.
```

Comment:
```suggestion
The protocol used a different format for keys and key bundles, see [PublicKeyBundle](https://github.com/xmtp/proto/blob/main/proto/message_contents/public_key.proto), although semantically there was very little difference. The old format had optional public key signatures, although this was never actually the case because the signatures were effectively required.
```